### DISCUSSION TITLE
`-work-dir` vs `-bucket-dir`

### DISCUSSION THREAD

**Discussion by bounlu** [nextflow-io/nextflow#3971] [Q&A]
Hi,

What is the difference between `-work-dir` and `-bucket-dir` when working with cloud i.e. `awsbatch`?

For example, either of these still store the intermediate result files at the specified S3 bucket:

```
-work-dir = 's3://nextflow/temp'
-bucket-dir = 's3://nextflow/temp'
```

**Comment by bentsherman** [comment]
I think it is only useful when you have some local tasks, then you use `-work-dir` for the local work directory and `-bucket-dir` for the remote work directory. The awsbatch executor will use the `-bucket-dir` first if it is defined.

**Comment by stevekm** [comment]
it seems like `-bucket-dir` is required if you want to use `nf-test` with AWS Batch execution? https://github.com/askimed/nf-test/issues/78
